core/src/main/java/org/bitcoinj/core/TransactionOutput.java

```diff
@@ -81,8 +82,8 @@ public class TransactionOutput {
      * @return read message
      * @throws BufferUnderflowException if the read message extends beyond the remaining bytes of the payload
      */
-    public static TransactionOutput read(ByteBuffer payload, Transaction parentTransaction) throws BufferUnderflowException, ProtocolException {
-        Objects.requireNonNull(parentTransaction);
+    public static TransactionOutput read(ByteBuffer payload, @Nullable Transaction parentTransaction)
+            throws BufferUnderflowException, ProtocolException {
```

Comment:
This makes me nervous, but I need to look more closely at this PR and consider alternatives before forming an actual opinion.